### DISCUSSION TITLE
Why does it generate many C language files when translating MO files into C language files?

### DISCUSSION THREAD

**Discussion by pily1** [OpenModelica/OpenModelica#12486] [Q&A] [answered]
hi，when I translate a mo file to C language file,it generate many C language files,why does it generate so many C lanugage files?
I just want use some code in embedded platform,how to control generate specific files? the following image is the output.
![1111](https://github.com/OpenModelica/OpenModelica/assets/20216269/e26b8c59-f7f8-479c-9d12-a8aeee39fd70)

**Comment by casella** [comment, marked as answer]
We generate many .c source code files so that they can be compiled in parallel, speeding up the compilation process. This should not be a problem, you just need to link them at the end of the process.
